File: Hash/위장.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> hm = new HashMap<>();
        for(String[] c: clothes){
            hm.put(c[1], hm.getOrDefault(c[1], 0)+1);
        }
        for(Map.Entry<String, Integer> el: hm.entrySet()){
            answer *= el.getValue()+1;
        }
        answer--;
        return answer;
    }
}

File: hash/위장.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, int> m;
    for(auto x : clothes) {
        m[x[1]]++;
    }
    
    for(auto x : m) {
        answer *= x.second + 1;
    }
    
    answer--;
    
    return answer;
}